internal/ansible/dispatcher/dispatcher_events.go

```diff
@@ -17,15 +16,36 @@ import (
 
 // Event-generating functions for the Playbook Dispatcher
 type EventList struct {
-	Events []message.AnsibleRunnerJobEventYaml
+	RunnerJobEvents []message.AnsibleRunnerJobEventYaml
+	Events          []results.AnsiblePlaybookJSONResultsPlay
 }
 
 func NewEventList() *EventList {
-	return &EventList{Events: []message.AnsibleRunnerJobEventYaml{}}
+	return &EventList{
+		RunnerJobEvents: []message.AnsibleRunnerJobEventYaml{},
```

Comment:
@machacekondra 
I'm not sure what to do here.
The `executor_on_start` and the `executor_on_failed` events are  `message.AnsibleRunnerJobEventYaml` type.
The ansible result events are `results.AnsiblePlaybookJSONResultsPlay`.

Am I correct? Is this a good way to replicate the rhc-worker behavior?

Reply:
I think you would need to write custom callback like this [one](https://github.com/alibaba/ansible-provider-docs/blob/master/lib/ansible/plugins/callback/json.py). And then parse the output as JSON.